react-validated-input-component/src/validated-input.jsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+
+export default class ValidatedInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: '',
+      error: '',
+      iconClass: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleValidation = this.handleValidation.bind(this);
+  }
+
+  handleChange(e) {
+    this.setState({ value: e.target.value });
+  }
+
+  handleValidation() {
```

Comment:
Based on how you are using this method in the `handleSubmit` this method should return `true` or `false` if the input is valid or not.